data/src/main/java/io/gnosis/data/repositories/TransactionRepository.kt

```diff
@@ -48,9 +55,35 @@ class TransactionRepository(
             )
         }
 
+    suspend fun getTransactionDetails(txId: String): TransferDetails =
+        gatewayApi.loadTransactionDetails(txId).let {
+
+            // TODO: create different tx types for detail display: Custom, Settingschange & Transfer
+
+            TransferDetails(
+                it.txHash,
+                it.txStatus,
+                (it.detailedExecutionInfo as? MultisigExecutionDetails)?.submittedAt?.toDate(),
+                it.executedAt?.toDate(),
+                getExecutor(it.txInfo),  //(it.txInfo as? Transfer)?.sender, // TODO: handle other transfer type
+                it.txData,
+                it.detailedExecutionInfo
+            )
+        }
+
+    private fun getExecutor(txInfo: TransactionInfo): Solidity.Address =
+         when (txInfo) {
+            is Custom -> txInfo.to // safe address ?
+            is SettingsChange -> "0x1234".asEthereumAddress()!! // safe address
```

Comment:
what are those values: "0x1234", ...? And they should be extracted to constants with meaningful names

Reply:
These are dummy values for now. They do not have any meaning. There is a TODO next to the call to getExecutor which will be covered in the next PR.

Reply:
this was not that clear. I would place TODO into getExecutor itself and not near it's invocation

Reply:
no action required now